commands/knowledge/arch.js

```diff
@@ -0,0 +1,133 @@
+const { MessageEmbed, MessageActionRow, MessageButton } = require('discord.js');
+const got = require('got');
+
+/**
+ * Converts a number of bytes to a human-readable string.
+ * @param {number} bytes The number of bytes.
+ * @param {number} [decimals=2] The number of decimal places to use.
+ * @returns {string} The human-readable string.
+ * @example
+ * bytesToSize(1024); // 1 KiB
+ */
+function bytesToSize(bytes, decimals = 2) {
+	const k = 1024;
+	const dm = decimals < 0 ? 0 : decimals;
+	const sizes = ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+
+	const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+	return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+}
+
+/**
+ * Modiefies the embed to display a certain number of packages.
+ * @param {MessageEmbed} embed The embed to modify.
+ * @param {Array} results The results to display.
+ * @param {number} page The page to display.
+ * @param {number} [n=5] The number of packages to display.
+ * @returns {MessageEmbed} The modified embed.
+ * @example
+ * const embed = new MessageEmbed();
+ *
+ * modifyEmbed(embed, results, 0); // Display the first 5 packages
+ */
+function modifyEmbed(embed, results, page, n = 5) {
+	const pages = Math.ceil(results.length / n);
+	const index = page * n;
+
+	embed.spliceFields(0, embed.fields.length);
+
+	for (let i = index; i < Math.min(index + n, results.length); i++) {
+		const {
+			arch,
+			compressed_size,
+			installed_size,
+			pkgdesc,
+			pkgname,
+			pkgver,
+			repo,
+		} = results[i];
+
+		const url = `https://www.archlinux.org/packages/${repo}/${arch}/${pkgname}/`;
+		const info = `${repo}/${pkgname} ${pkgver} (${bytesToSize(compressed_size)} ${bytesToSize(installed_size)})`;
+
+		const name = `${i + 1}. ${info}`;
+		const value = `${pkgdesc}\n${url}`;
+
+		embed.addField(name, value);
+	}
+
+	embed.setFooter({ text: `Page ${page + 1} of ${pages} | ${results.length} results` });
+
+	return embed;
+}
+
+
+// TODO: aur support
+module.exports = {
+	name: 'arch',
+	description: 'Queries Arch Linux Packages.',
+	usage: 'arch <query>',
+	args: true,
+	disabled: false,
+	aliases: ['pacman', 'yay'],
```

Comment:
```suggestion
	aliases: ['pacman', 'yay', 'aur'],
```

Reply:
I plan to add that alias once aur search is actually implemented.

Reply:
oh lol. salad's working on that right?